uni/lib/view/library/library.dart

```diff
@@ -1,126 +1,75 @@
 import 'package:flutter/material.dart';
-import 'package:percent_indicator/linear_percent_indicator.dart';
-import 'package:provider/provider.dart';
 import 'package:uni/generated/l10n.dart';
-import 'package:uni/model/entities/library_occupation.dart';
-import 'package:uni/model/providers/lazy/library_occupation_provider.dart';
-import 'package:uni/utils/drawer_items.dart';
 import 'package:uni/view/common_widgets/page_title.dart';
 import 'package:uni/view/common_widgets/pages_layouts/general/general.dart';
-import 'package:uni/view/lazy_consumer.dart';
-import 'package:uni/view/library/widgets/library_occupation_card.dart';
+import 'package:uni/view/library/widgets/library_occupation_tab.dart';
+import 'package:uni/view/library/widgets/library_reservations_tab.dart';
 
 class LibraryPage extends StatefulWidget {
-  const LibraryPage({super.key});
+  const LibraryPage({this.startOnOccupationTab = true, super.key});
+  final bool startOnOccupationTab;
 
   @override
   State<StatefulWidget> createState() => LibraryPageState();
 }
 
 class LibraryPageState extends GeneralPageViewState<LibraryPage> {
-  @override
-  Widget getBody(BuildContext context) {
-    return ListView(
-      shrinkWrap: true,
-      children: [
-        PageTitle(name: S.of(context).nav_title(DrawerItem.navLibrary.title)),
-        LibraryOccupationCard(),
-        PageTitle(name: S.of(context).floors),
-        LazyConsumer<LibraryOccupationProvider, LibraryOccupation>(
-          builder: getFloorRows,
-          hasContent: (occupation) => occupation.floors.isNotEmpty,
-          onNullContent: Center(
-            child: Text(
-              S.of(context).no_library_info,
-              style: const TextStyle(fontSize: 18),
-            ),
-          ),
-          contentLoadingWidget: const CircularProgressIndicator(),
-        ),
-      ],
-    );
-  }
+  late TabController tabController;
+  late List<Tab> tabs;
+  static const LibraryOccupationTab _libraryOccupationTab =
+      LibraryOccupationTab();
+  static const LibraryReservationsTab _libraryReservationTab =
+      LibraryReservationsTab();
```

Comment:
How do we need this? Can't we just the tabs in the array and init the array properly below?